src/components/maps/CameraViewer.jsx

```diff
@@ -0,0 +1,188 @@
+import PropTypes from 'prop-types';
+import { Actions } from 'jumpstate';
+
+import Anchor from 'grommet/components/Anchor';
+import Box from 'grommet/components/Box';
+import Button from 'grommet/components/Button';
+import Label from 'grommet/components/Label';
+import Layer from 'grommet/components/Layer';
+import List from 'grommet/components/List';
+import ListItem from 'grommet/components/ListItem';
+import Notification from 'grommet/components/Notification';
+import Tile from 'grommet/components/Tile';
+import Tiles from 'grommet/components/Tiles';
+import FormNextIcon from 'grommet/components/icons/base/FormNext';
+import FormPreviousIcon from 'grommet/components/icons/base/FormPrevious';
+import SpinningIcon from 'grommet/components/icons/Spinning';
+
+import {
+  MAPEXPLORER_INITIAL_STATE, MAPEXPLORER_PROPTYPES,
+  MAPEXPLORER_CAMERA_STATUS,
+} from '../../ducks/mapexplorer';
+
+const ITEMS_PER_PAGE = 6;
+const THUMBNAIL_SERVER_URL = 'https://thumbnails.zooniverse.org/';
+const THUMBNAIL_WIDTH = 320;
+const THUMBNAIL_HEIGHT = 200;
+
+class CameraViewer extends React.Component {
+  constructor(props) {
+    super(props);
+  
+    this.renderData = this.renderData.bind(this);
+    this.renderMetadata = this.renderMetadata.bind(this);
+    this.renderDataPaging = this.renderDataPaging.bind(this);
+    this.changeDataPaging = this.changeDataPaging.bind(this);
+  
+    this.state = {
+      page: 0,
+    };
+  }
+
+  componentDidMount() {
+    this.setState({
+      page: 0,
+    });
+  }
+  
+  render() {
+    if (!this.props.mapConfig) return null;
+    
+    return (
+      <Layer className="camera-viewer">
+        <Box className="content" align="center">
+          <Box className="camera-metadata">
+            {this.renderMetadata()}
+          </Box>
+          <Box className="camera-data">
+            {this.renderDataPaging()}
+            {this.renderData()}
+          </Box>
+          <Button label="Done" onClick={()=>{ Actions.mapexplorer.resetActiveCamera() }}></Button>
+        </Box>
+      </Layer>
+    );
+  }
+
+  renderMetadata() {
+    if (this.props.activeCameraMetadataStatus === MAPEXPLORER_CAMERA_STATUS.FETCHING) {
+      return <SpinningIcon />;
+    } else if (this.props.activeCameraMetadataStatus === MAPEXPLORER_CAMERA_STATUS.ERROR) {
+      return <Notification message='ERROR' status='critical' />
+    }
+    
+    if (this.props.activeCameraMetadataStatus === MAPEXPLORER_CAMERA_STATUS.SUCCESS &&
+        this.props.activeCameraMetadata) {
+
+      return (
+        <List>
+        {Object.keys(this.props.activeCameraMetadata).map(key => (
+          <ListItem
+            key={`camera-metadata-${key}`}
+            justify='between'
+            separator='horizontal'
+          >
+            <span>{key}</span>
+            <span>{this.props.activeCameraMetadata[key]}</span>
+          </ListItem>
+        ))}
+        </List>
+      );
+    }
+
+    return null;
+  }
+
+  renderData() {
```

Comment:
Maybe this can be a functional child component?

Reply:
👌